pkg/controllers/cassandra/service/control.go

```diff
@@ -34,9 +37,17 @@ func NewControl(
 
 func (e *defaultCassandraClusterServiceControl) Sync(cluster *v1alpha1.CassandraCluster) error {
 	svc := ServiceForCluster(cluster)
-	_, err := e.kubeClient.CoreV1().Services(svc.Namespace).Update(svc)
+	client := e.kubeClient.CoreV1().Services(svc.Namespace)
+	existingSvc, err := e.serviceLister.Services(cluster.Namespace).Get(svc.Name)
 	if k8sErrors.IsNotFound(err) {
-		_, err = e.kubeClient.CoreV1().Services(svc.Namespace).Create(svc)
+		_, err := client.Create(svc)
+		return err
+	} else if err != nil {
+		return err
+	}
+	if !metav1.IsControlledBy(existingSvc, cluster) {
+		return fmt.Errorf("Duplicate service detected: %s", svc)
```

Comment:
This check looks good to me.

Can we update the error message to state that an existing service of name xyz already exists, and that we won't touch it? (so that users know they need to intervene). Also `%s` with `svc` (which is of type `v1.Service`) will give some ugly output that isn't useful for users (it'll be the address of the pointer). Can we update to be the service name? (`svc.Name`). It's the service name that has caused the conflict here, and it's by name that a user will fix this problem 😄

Reply:
Good point. Done.